service/controller/clusterapi/v30/resource/tcnp/create.go

```diff
@@ -333,7 +334,7 @@ func newLaunchConfiguration(ctx context.Context, cr v1alpha1.MachineDeployment)
 			Type:       key.MachineDeploymentInstanceType(cr),
 		},
 		SmallCloudConfig: template.ParamsMainLaunchConfigurationSmallCloudConfig{
-			S3URL: key.SmallCloudConfigS3URL(&cr, cc.Status.TenantCluster.AWS.AccountID, "worker"),
+			S3URL: fmt.Sprintf("s3://%s/%s", key.BucketName(&cr, cc.Status.TenantCluster.AWS.AccountID), key.S3ObjectPathTCNP(&cr)),
```

Comment:
Why do we use `fmt.Sprintf` here? I have difficulty grasping what this path returns now / if it's accurate because we have key functions for everything else.

Reply:
The alternative wasn't better. Having a specific function for TCCP and TCNP and injecting a lot of parameters. This here made sense to me.